apis/python/src/tiledbsoma/types.py

```diff
@@ -16,7 +16,7 @@
 BatchFormat = Literal["dense", "coo", "csr", "csc", "record-batch", "table"]
 ReadPartitions = Literal["IofN"]
 BatchSize = Literal["count", "size", "auto"]
-ResultOrder = Literal["row-major", "column-major", "unordered", "rowid-ordered"]
+ResultOrder = Literal["row-major", "column-major", "auto"]
```

Comment:
@bkmartinjr @johnkerl I didn't see an obvious way to implement "rowid-ordered" by pushing down into TileDB; care to correct me on that? Is it important enough to support to either (i) sort ourselves in-memory or (ii) detect if soma_joinid is the first or second indexed dimension and set row-major or column-major accordingly?

General background question...say we've indexed three or more dimensions...are the "rows" and "columns" simply the first two, as far as TileDB is concerned? No way to ask for results ordered by the third or higher dimension? (I think I can guess reasons for this based on the first two dimensions determining the on-disk layout, but just curious if something else distinguishes the two dimensions "rows" and "columns")

Reply:
> "rowid-ordered"

Correct - this is unsupported for anything other than DenseNdArray (or the original "dense" offset-indexed dataframe, which is scuttled for now).

> General background question

Nope..."row" is a synonym for C order, "col" for Fortran order.  All dimensions (ie. the entire hypercube) are ordered, either last dimension to first dimension (C or row order), or first dimension to last dimension (Fortran or column order)